src/checkInternetConnectionOnStartup.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+
+import { Platform, NetInfo } from 'react-native';
+import checkInternetAccess from './checkInternetAccess';
+
+// on iOS, the listener is fired immediately after registration
+// on Android, we need to use `isConnected.fetch`, that returns a promise which resolves with a boolean
+export default function checkInternetConnectionOnStartup(
+  timeout: number = 3000,
+  url: string = 'https://google.com',
+): Promise<boolean> {
+  let connectionChecked: Promise<boolean>;
+  if (Platform.OS === 'ios') {
+    connectionChecked = new Promise(resolve: Function => {
```

Comment:
There is a `flow` error, could you please wrap `(resolve: Function)` into parenthesis?

I don't know if that's due to an old version (v0.42). Another alternative could be to upgrade `flow` to a most recent one.

But probably the easiest is the former